src/builder/BridgeRoutes.sol

```diff
@@ -9,61 +9,88 @@ library BridgeRoutes {
+        address outgoingAddress;
         BridgeType bridgeType;
     }
 
-    function hasBridge(uint256 srcChainId, uint256 dstChainId, string memory assetSymbol) internal pure returns (bool) {
+    function hasBridge(uint256 srcChainId, uint256 dstChainId, string memory assetSymbol)
+        internal
+        pure
+        returns (bool)
+    {
         if (getBridge(srcChainId, dstChainId, assetSymbol).bridgeType == BridgeType.NONE) {
             return false;
         } else {
             return true;
         }
     }
 
-    function getBridge(uint256 srcChainId, uint256 dstChainId, string memory assetSymbol) internal pure returns (Bridge memory) {
-        if (srcChainId == 1) {
-            return getBridgeForMainnet(dstChainId, assetSymbol);
-        } else if (srcChainId == 8453) {
-            return getBridgeForBase(dstChainId, assetSymbol);
+    // TODO: actually consider dstChainId before returning a bridge?
+    // TODO: resolve assetSymbol to address before trying to bridge?
+    // TODO: how do we represent the constant information about existing bridges? without mappings, etc.
+    function getBridge(uint256 srcChainId, uint256 /*dstChainId*/, string memory assetSymbol)
+        internal
+        pure
+        returns (Bridge memory bridge)
+    {
+        if (CCTP.canBridge(srcChainId, assetSymbol)) {
+            bridge = Bridge({
+                chainId: srcChainId,
+                outgoingAddress: CCTP.knownOutgoingAddress(srcChainId),
+                bridgeType: BridgeType.CCTP
+            });
+        }
+    }
+}
+
+// NOTE: this library would be generated by a network crawler script.
+library CCTP {
+    // IDEA: input this into codegen, get out knownDomainId and knownOutgoingAddress functions?
+    struct Roots {
+        uint256 chainId;
+        uint256 domainId;
+        address outgoingAddress;
+        uint256 estimatedCost;
+    }
+
+    error NoKnownBridge(uint256 chainId);
+    error MissingDomainId(uint256 chainId);
+
+    function knownDomainId(uint256 chainId) internal pure returns (uint32) {
+        if (chainId == 1) {
+            return 0;
+        } else if (chainId == 8453) {
+            return 6;
         } else {
-            return Bridge({
-                    bridgeAddress: address(0),
-                    bridgeType: BridgeType.NONE
-                });
-            // revert BridgeNotFound(1, dstChainid, assetSymbol);
+            revert MissingDomainId(chainId);
         }
     }
 
-    function getBridgeForMainnet(uint256 /*dstChainId*/, string memory assetSymbol) internal pure returns (Bridge memory) {
-        if (Strings.stringEqIgnoreCase(assetSymbol, "USDC")) {
-            return Bridge({
-                bridgeAddress: 0xBd3fa81B58Ba92a82136038B25aDec7066af3155,
-                bridgeType: BridgeType.CCTP
-            });
+    function estimatedCost(uint256 chainId) internal pure returns (uint256) {
+        if (chainId == 1) {
+            return 1_000_000;
```

Comment:
I'm not entirely sure what this is supposed to represent; :magic_wand:  could probably clarify.

I will probably remove `estimatedCost`, and we can just add it back later if we need it.

Reply:
Let's remove it for now. 
We can leave this for later time when we want to improve the bridge actions. 

Main purpose is to avoid from the scenario when it try to spend $5 in eth transaction cost on mainnet to bridge $0.25 to Base.

Reply:
This has been removed